.github/workflows/ci-mingw.yml

```diff
@@ -0,0 +1,89 @@
+# This is a copy of ci-unix-shared-local.yml for building shared libraries
+# with an additional build configuration (using installed deps and dav1d).
+
+name: CI MINGW
+on: [push, pull_request]
+
+permissions:
+  contents: read
+
+# Cancel the workflow if a new one is triggered from the same PR, branch, or tag, except on main.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  build-mingw:
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        sys:
+          - mingw64
+          - mingw32
+          - ucrt64
+          - clang64
+
+    steps:
+    - uses: msys2/setup-msys2@v2
+      with:
+        msystem: ${{matrix.sys}}
+        update: true
+        install: >-
+          aom
+          cmake
+          dav1d
+          gcc
+          git
+          libyuv
+          ninja
+    - name: Cache external dependencies
+      id: cache-ext
+      uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1
+      with:
+        path: ext
+        key: ${{ runner.sys }}-mingw
+    - name: Print cmake version
+      run: cmake --version
+    - name: Build GoogleTest
+      if: steps.cache-ext.outputs.cache-hit != 'true'
+      working-directory: ./ext
+      run: bash -e googletest.cmd
```

Comment:
Just in case you want to make the CI faster, MSYS2 ships [googletest](https://packages.msys2.org/base/mingw-w64-googletest) already...

Reply:
Right, but I thought (just like on Ubuntu), that it was better to compile gtest from source to make sure the same compile flags are passed?